js/src/Themes.js

```diff
@@ -50,10 +55,35 @@ export class ThemeModel extends WidgetModel {
     } else if (document.body.classList.contains("theme-light")) {
       this.set("dark", false);
       this.save_changes();
+    } else if (window.Jupyter) {
+      // Special case for Jupyter Notebook
+      this.set("dark", false);
+      this.save_changes();
+    } else if (document.body.dataset.vscodeThemeKind === "vscode-dark") {
+      // Special case for VS Code
+      vuetify.framework.theme.dark = true;
+      this.set("dark", true);
+      this.save_changes();
+    } else if (document.body.dataset.vscodeThemeKind === "vscode-light") {
+      vuetify.framework.theme.dark = false;
+      this.set("dark", false);
+      this.save_changes();
```

Comment:
So I think if dark==null signals 'auto', we should only set "dark_effective", and not "dark" right?

Also, should we listen to changes in the property?

Reply:
(Assuming we can listen to attribute changes)

Reply:
Ah, you're correct about it having to be `dark_effective`.

Reply:
When you say

> should we listen to changes in the property?

You mean `dark_effective`?